storage-proofs-update/src/vanilla.rs

```diff
@@ -0,0 +1,1172 @@
+    pub challenge_bit_len: usize,
+    // The number of partition proofs for this sector-size.
+    pub partition_count: usize,
+    // The bit length of an integer in `0..partition_count` which is also the height of the
+    // partitions-tree within TreeD, i.e. the top of TreeD starting from the tree row containing
+    // each partition's apex-root and ending at TreeD's root.
+    pub partition_bit_len: usize,
+    // The number of leafs in the apex-tree.
+    pub apex_leaf_count: usize,
+    // The bit length of an integer in `0..apex_leaf_count` which is also the height of each
+    // partition's apex-tree.
+    pub apex_leaf_bit_len: usize,
+}
+
+impl ParameterSetMetadata for PublicParams {
+    fn identifier(&self) -> String {
+        format!(
+            "empty_sector_update::PublicParams{{ sector_nodes: {} }}",
+            self.sector_nodes
+        )
+    }
+
+    fn sector_size(&self) -> u64 {
+        (self.sector_nodes as u64) << 5
+    }
+}
+
+impl PublicParams {
+    pub fn from_sector_size(sector_bytes: u64) -> Self {
+        // The sector-size measured in 32-byte nodes.
+        let sector_nodes = ALLOWED_SECTOR_SIZES
+            .iter()
+            .copied()
+            .find(|allowed_nodes| (allowed_nodes << 5) as u64 == sector_bytes)
+            .expect("provided sector-size is not allowed");
+
+        // `sector_nodes` is guaranteed to be a power of two.
+        let challenge_bit_len = sector_nodes.trailing_zeros() as usize;
+        let challenge_count = challenge_count(sector_nodes);
+
+        let partition_count = partition_count(sector_nodes);
+        // `partition_count` is guaranteed to be a power of two.
+        let partition_bit_len = partition_count.trailing_zeros() as usize;
+
+        let apex_leaf_count = apex_leaf_count(sector_nodes);
+        // `apex_leaf_count` is guaranteed to be a power of two.
+        let apex_leaf_bit_len = apex_leaf_count.trailing_zeros() as usize;
+
+        PublicParams {
+            sector_nodes,
+            challenge_count,
+            challenge_bit_len,
+            partition_count,
+            partition_bit_len,
+            apex_leaf_count,
+            apex_leaf_bit_len,
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct PublicInputs {
+    pub k: usize,
+    pub comm_r_old: TreeRDomain,
+    pub comm_d_new: TreeDDomain,
+    pub comm_r_new: TreeRDomain,
+    // The number of high bits to take from each challenge's bits. Used to verify replica encoding
+    // in the vanilla proof. `h` is only a public-input for the vanilla proof; the circuit takes
+    // `h_select` as a public-input rather than `h`.
+    pub h: usize,
+}
+
+pub struct PrivateInputs {
+    pub comm_c: TreeRDomain,
+    pub tree_r_old_config: StoreConfig,
+    // Path to old replica.
+    pub old_replica_path: PathBuf,
+    pub tree_d_new_config: StoreConfig,
+    pub tree_r_new_config: StoreConfig,
+    // Path to new replica.
+    pub replica_path: PathBuf,
+}
```

Comment:
Do all members (also for the `PublicParams` above) need to be public? I try to keep as much out of the public API as possible. Once somrthing is in the public API, it's tricky to get it out again. E.g. If you want to change a type (let's say you change from `PathBuf` to `Path`) and it's public, it's a breaking change and you have to be very careful about it.